### PR TITLE
I added dropzone for image support as input

### LINKED ISSUE #67: Add support for image as inputs in chat
Move to gpt-4-turbo vision model

### PR DESCRIPTION
**Pull request by Rohittomar01**
Will Close #67

### CODE REVIEW

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:23]: Here is `File`
```suggestion
  if (file && file instanceof File) {
```
Here it is

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:13]: send whole messages array to the api for full chat context

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:40]: use saveToDB from utils/apiHelpers to save to db and post to algolia https://github.com/Technoculture/echoes/blob/dev/src/utils/apiHelper.ts

File: src/middleware.ts
Comment (PrinceBaghel258025) [inline src/middleware.ts:10]: no need to ignore any route.
don't even need to interact with middleware.ts

File: src/components/inputBar.tsx
Comment (PrinceBaghel258025) [inline src/components/inputBar.tsx:474]: having only one button can do the thing,  no need to have two different buttons rendering conditionally

File: src/components/inputBar.tsx
Comment (PrinceBaghel258025) [inline src/components/inputBar.tsx:451]: Instead of calling this callback here, call it inside the handleSubmit and it won't require you to have an explicit state for imageInput

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts]: Resolve it

File: src/middleware.ts
Comment (Rohittomar01) [reply src/middleware.ts]: Resolved it

**Comment by sutyum** [comment]
There is no image visible in the chat?

File: src/components/inputBar.tsx
Comment (Rohittomar01) [reply src/components/inputBar.tsx]: Resolved it

File: src/components/inputBar.tsx
Comment (Rohittomar01) [reply src/components/inputBar.tsx:545]: resolved it

File: src/components/inputBar.tsx
Comment (sutyum) [inline src/components/inputBar.tsx:38]: dropZoneImages instead as its a list

File: src/lib/types/ai.d.ts
Comment (sutyum) [inline src/lib/types/ai.d.ts:6]: use "input-image" instead

File: src/components/inputBar.tsx
Comment (sutyum) [inline src/components/inputBar.tsx:125]: zod

File: src/app/api/chatmodel/[chatid]/route.ts
Comment (sutyum) [inline src/app/api/chatmodel/[chatid]/route.ts:29]: ?

**Review by sutyum — CHANGES_REQUESTED** [review]

**Review by sutyum — CHANGES_REQUESTED** [review]

File: src/app/api/imageInput/route.ts
Comment (sutyum) [reply src/app/api/imageInput/route.ts:16]: @Rohittomar01

File: src/components/DropzoneDrawer.tsx
Comment (sutyum) [inline src/components/DropzoneDrawer.tsx:84]: why not using tailwind here?

File: src/components/DropzoneDrawer.tsx
Comment (sutyum) [inline src/components/DropzoneDrawer.tsx:75]: why this styling override?

File: src/components/DropzoneDrawer.tsx
Comment (sutyum) [inline src/components/DropzoneDrawer.tsx:71]: <Button size="icon" variant="disabled">
instead of
<Button size="icon" variant="secondary" className="disabled:text-muted">

**Review by sutyum — CHANGES_REQUESTED** [review]

File: src/app/api/imageInput/route.ts
Comment (sutyum) [reply src/app/api/imageInput/route.ts:11]: You can remove this live in order to use the default maxTokens value

File: src/components/chat.tsx
Comment (Rohittomar01) [reply src/components/chat.tsx:39]: Corrected it

File: src/components/inputBar.tsx
Comment (Rohittomar01) [reply src/components/inputBar.tsx:157]: used zod here

File: src/components/inputBar.tsx
Comment (Rohittomar01) [reply src/components/inputBar.tsx:58]: resolved it

File: src/components/chat.tsx
Comment (Rohittomar01) [reply src/components/chat.tsx:45]: handled it

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts]: used zod here

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts]: used zod here

File: src/app/api/chatmodel/[chatid]/route.ts
Comment (Rohittomar01) [reply src/app/api/chatmodel/[chatid]/route.ts]: resolved it

File: src/components/DropzoneDrawer.tsx
Comment (Rohittomar01) [reply src/components/DropzoneDrawer.tsx]: resolved it

File: src/components/DropzoneDrawer.tsx
Comment (Rohittomar01) [reply src/components/DropzoneDrawer.tsx]: resolved it

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:119]: why do we need this ?

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:151]: can't we do proper streaming instead of faking it on frontend

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:109]: this still is not correct

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:91]: we decided not to use a single message instead you have to feed all the message to model to get proper context.

**Review by mprinceb — CHANGES_REQUESTED** [review]

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts]: have sent

File: next.config.mjs
Comment (Rohittomar01) [reply next.config.mjs:25]: old code

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts:20]: I can understand it

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts]: older one

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts]: older one

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts]: older one

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts]: older one

File: next.config.mjs
Comment (PrinceBaghel258025) [reply next.config.mjs:25]: but why do we need to allow images from this domain ?

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:76]: iinstead of having a seperate function extend the previous one(jsonToLangchain) if things are working

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:97]: save dropped image in this callback

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:159]: move this to onStart callback

File: src/components/chatmessage.tsx
Comment (PrinceBaghel258025) [inline src/components/chatmessage.tsx:22]: this would have been from shadcn (our local ui components)

File: src/components/chatmessage.tsx
Comment (PrinceBaghel258025) [inline src/components/chatmessage.tsx:242]: why do we need to call a function to render jsx ?

File: src/components/chatmessagecombinator.tsx
Comment (PrinceBaghel258025) [inline src/components/chatmessagecombinator.tsx:168]: why this have been change to return only subrole ?

File: src/lib/types/ai.d.ts
Comment (PrinceBaghel258025) [inline src/lib/types/ai.d.ts:6]: but it still is "image"

File: src/components/inputBar.tsx
Comment (PrinceBaghel258025) [inline src/components/inputBar.tsx:193]: can not this new messages be saved to db in onCompletion callback of LangchainStream ?

File: next.config.mjs
Comment (Rohittomar01) [reply next.config.mjs:25]: It needed before public aws bucket.Now have removed it .

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts:97]: Resolved

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts:147]: No

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts:159]: Resolved

File: src/components/chatmessagecombinator.tsx
Comment (Rohittomar01) [reply src/components/chatmessagecombinator.tsx:168]: It written for try to  check the patent search bug but now I removed all extra things and it made as it was before.

File: src/components/chatmessage.tsx
Comment (PrinceBaghel258025) [inline src/components/chatmessage.tsx:70]: remove commented code

File: src/components/inputBar.tsx
Comment (PrinceBaghel258025) [inline src/components/inputBar.tsx:192]: this api call is not required at all, everything to construct the newer ai message is available in onCompletion callback

File: src/components/inputBar.tsx
Comment (PrinceBaghel258025) [inline src/components/inputBar.tsx:347]: remove it

File: src/components/inputBar.tsx
Comment (PrinceBaghel258025) [inline src/components/inputBar.tsx:248]: remove it

File: src/components/inputBar.tsx
Comment (PrinceBaghel258025) [inline src/components/inputBar.tsx:219]: remove it

File: src/components/rendermarkdown.tsx
Comment (PrinceBaghel258025) [inline src/components/rendermarkdown.tsx:96]: remove it

File: src/utils/apiHelper.ts
Comment (PrinceBaghel258025) [inline src/utils/apiHelper.ts:98]: remove it

File: src/utils/apiHelper.ts
Comment (PrinceBaghel258025) [inline src/utils/apiHelper.ts:425]: remove it

File: src/components/rendermarkdown.tsx
Comment (PrinceBaghel258025) [inline src/components/rendermarkdown.tsx:96]: remove it

File: src/components/inputBar.tsx
Comment (PrinceBaghel258025) [inline src/components/inputBar.tsx:347]: remove it

**Review by mprinceb — CHANGES_REQUESTED** [review]

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [inline src/app/api/imageInput/route.ts:107]: why is this commented ?
we decided to remove that extra call from frontend side not from here

File: next.config.mjs
Comment (PrinceBaghel258025) [inline next.config.mjs:27]: this is the prefix url having https:// prefix with it

File: src/app/api/imageInput/route.ts
Comment (PrinceBaghel258025) [reply src/app/api/imageInput/route.ts:23]:
```suggestion
  chatId: z.coerce.number(),
```

File: src/app/api/imageInput/route.ts
Comment (Rohittomar01) [reply src/app/api/imageInput/route.ts]: Resolved

File: src/components/inputBar.tsx
Comment (Rohittomar01) [reply src/components/inputBar.tsx]: resolved it

File: src/components/DropzoneDrawer.tsx
Comment (sutyum) [inline src/components/DropzoneDrawer.tsx:49]: ?

**Review by sutyum — CHANGES_REQUESTED** [review]

File: src/app/api/imageInput/route.ts
Comment (sutyum) [inline src/app/api/imageInput/route.ts:82]: why 2024?

**Review by sutyum — CHANGES_REQUESTED** [review]

**Comment by sutyum** [comment]
@PrinceBaghel258025 waiting for your final review / go ahead - in order to merge this PR

**Review by mprinceb — APPROVED** [review]

**Pull request merged**
